src/uu/pkill/src/pkill.rs

```diff
@@ -71,25 +73,53 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
 
 #[cfg(unix)]
 fn handle_obsolete(args: &mut [String]) {
-    // Sanity check
-    if args.len() > 2 {
-        // Old signal can only be in the first argument position
-        let slice = args[1].as_str();
-        if let Some(signal) = slice.strip_prefix('-') {
+    for arg in &mut args[1..] {
+        if let Some(signal) = arg.strip_prefix('-') {
             // Check if it is a valid signal
             let opt_signal = signal_by_name_or_value(signal);
             if opt_signal.is_some() {
                 // Replace with long option that clap can parse
-                args[1] = format!("--signal={signal}");
+                *arg = format!("--signal={signal}");
             }
         }
     }
 }
 
+// Not contains in libc
+#[cfg(target_os = "linux")]
+extern "C" {
+    fn sigqueue(
+        pid: uucore::libc::pid_t,
+        sig: uucore::libc::c_int,
+        val: uucore::libc::sigval,
+    ) -> uucore::libc::c_int;
+}
+
 #[cfg(unix)]
-fn kill(pids: &Vec<ProcessInformation>, sig: Option<Signal>, echo: bool) {
+#[allow(unused_variables)]
+fn kill(pids: &Vec<ProcessInformation>, sig: Option<Signal>, queue: Option<u32>, echo: bool) {
     for pid in pids {
-        if let Err(e) = signal::kill(Pid::from_raw(pid.pid as i32), sig) {
+        #[cfg(target_os = "linux")]
+        let result = if let Some(queue) = queue {
+            let v = unsafe {
+                sigqueue(
+                    pid.pid as i32,
+                    match sig {
+                        Some(s) => s as uucore::libc::c_int,
+                        None => 0,
+                    },
```

Comment:
I think you can simplify this `match` with `map_or`:
```suggestion
                    sig.map_or(0, |s| s as uucore::libc::c_int),
```